homeassistant/components/rainforest_eagle/sensor.py

```diff
@@ -49,14 +50,26 @@
 )
 
 
+def hwtest(cloud_id, install_code, ip_address):
+    """Try API call 'device_list' to see if target device is Legacy or Eagle-200."""
+    reader = LeagleReader(cloud_id, install_code, ip_address)
+    response = reader.post_cmd("device_list")
+    if "Error" in response and "Unknown command" in response["Error"]["Text"]:
+        return reader  # Probably a Legacy model
+    if "DeviceList" in response:
+        return EagleReader(ip_address, cloud_id, install_code)  # Probably Eagle-200
+
+    _LOGGER.error("Couldn't determine device model.")
```

Comment:
Don't log an error but raise so set up is aborted.
```suggestion
    raise ValueError("Couldn't determine device model.")
```